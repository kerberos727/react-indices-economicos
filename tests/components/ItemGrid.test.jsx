const { render, screen } = require("@testing-library/react");
const { ItemGrid } = require("../../src/components/ItemGrid");

const mockIndex = [
  {
    indice: {
      codigo: "uf",
      nombre: "Unidad de fomento (UF)",
      unidad_medida: "Pesos",
      fecha: "2022-08-20T04:00:00.000Z",
      valor: 33669,
    },
  },
];

describe("Tests in ItemGrid Component", () => {
  it("match with snapshot", () => {
    const index = mockIndex[0].indice;
    const { container } = render(<ItemGrid indice={index} />);
    expect(container).toMatchSnapshot();
  });

  it("should show title of indice in component", () => {
    const index = mockIndex[0].indice;
    render(<ItemGrid indice={index} />);
    expect(screen.getByText(index.nombre)).toBeTruthy();
  });
});
