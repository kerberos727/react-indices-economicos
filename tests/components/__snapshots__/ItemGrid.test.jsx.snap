// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in ItemGrid Component match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="title"
      data-testid="title"
    >
      Unidad de fomento (UF)
    </div>
    <div
      class="value"
    >
      33,669
    </div>
    <div
      class="detail"
    >
      Pesos
      <br />
      Sat Aug 20 2022
    </div>
  </div>
</div>
`;
